### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by Khareayushh**
## What kind of change does this PR introduce?

Docs Update ( specifically in quickstart.mdx).

This pull request updates the documentation for invoking Edge Functions in the Supabase API by adding a new section that covers error handling during function invocations. The goal is to provide users with a comprehensive guide on how to handle potential errors and exceptions that may arise when interacting with Edge Functions. This enhances the overall robustness of the documentation and improves the user experience.

## What is the current behavior?

The current documentation provides instructions on how to invoke Edge Functions using cURL and the "supabase-js" client library. It covers the process of invoking functions, including making POST requests with required headers and data. However, it lacks information on how to handle errors that may occur during these invocations.

The link to the issue ( https://github.com/supabase/supabase/issues/16856 )

## What is the new behavior?

The updated documentation retains the instructions for invoking Edge Functions using both cURL and the "supabase-js" client library. In addition to the invocation process, the documentation now includes a new section dedicated to error handling. This section provides code examples and explanations on how to effectively handle different types of errors that might occur during the invocation process. The "supabase-js" code snippet has been expanded to include an error handling mechanism that differentiates between FunctionsHttpError, FunctionsRelayError, and FunctionsFetchError. This update enhances the documentation by addressing potential issues that users might face and providing solutions for handling errors, ultimately contributing to a more robust user experience.

Feel free to include screenshots if it includes visual changes.

## Additional context

The previous version of the documentation lacked information on how to handle errors that can occur during Edge Function invocations. This update addresses that gap by introducing a new section dedicated to error handling. The previous version only covered the process of invoking functions and receiving responses. The current version enhances the documentation by including comprehensive instructions for dealing with potential errors, thus ensuring a smoother user experience and better-equipped developers.

Add any other context or screenshots.

### CODE REVIEW

**Comment by saltcod** [comment]
Hey @Khareayushh — curious what's up with this PR. It's absolutely massive! Must be the wrong base?

**Comment by Khareayushh** [comment]
After creating this PR, I have checked this branch is much more lagging behind.
I have also created a PR to the master branch, is it correct?

**Comment by saltcod** [comment]
Aah gotcha. Bringing it up to date should reduce the changes then. Cool. 

(and yep — master is the branch!)

**Comment by laktek** [comment]
@Khareayushh Looks cool! Thanks for the contribution. cc: @thorwebdev

File: apps/docs/pages/guides/functions/quickstart.mdx [hunk 1]
```diff
@@ -106,6 +106,33 @@ const { data, error } = await supabase.functions.invoke('hello-world', {
...
+const { data, error } = await supabase.functions.invoke('hello', {
+  headers: {
+    "my-custom-header": 'my-custom-header-value'
+  },
+  body: { foo: 'bar' }      
+})
+
+if (error instanceof FunctionsHttpError) {
+  const errorMessage = await error.context.json()
+  console.log('Function returned an error', errorMessage)
```
Comment (laktek) [inline apps/docs/pages/guides/functions/quickstart.mdx:125]: minor: use `console.error`?

**Review by laktek — APPROVED** [review]

**Review by kevcodez — APPROVED** [review]

File: apps/docs/pages/guides/functions/quickstart.mdx (hunk 1 above)
Comment (Khareayushh) [reply apps/docs/pages/guides/functions/quickstart.mdx:125]: I think @kevcodez resolved this console.error.

Comment (kevcodez) [reply apps/docs/pages/guides/functions/quickstart.mdx:125]: I didn't - it's still console.log right now

**Comment by thorwebdev** [comment]
@Khareayushh  I'm sorry that this is such a pain, but could you please run prettier?

**Comment by Khareayushh** [comment]
@thorwebdev I have applied prettier in quickstart.mdx

**Comment by thorwebdev** [comment]
@Khareayushh I'm afraid the prettier check is still failing :( 

I really do want you to get the attribution for this, so maybe could you start from a fresh fork from the master branch and open a new PR and link it here.

Alternatively I can open a PR with your changes myself. I can see if I can get you listed as a contributor for that, but I'm not sure. 

Let me know if you have any preference. Sorry again that this has been such a painfully long experience!

**Comment by Khareayushh** [comment]
@thorwebdev can you help me by suggesting how to apply prettier? I have applied prettier by installing it with npm install and then run prettier by going to quickstart.mdx.

**Comment by Khareayushh** [comment]
Let me do it again.

File: package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
   "devDependencies": {
     "eslint": "^8.41.0",
     "eslint-config-custom": "*",
-    "prettier": "^2.8.8",
+    "prettier": "2.8.8",
```
Comment (thorwebdev) [inline package.json:44]: Please don't modify the package.json!

**Review by thorwebdev — CHANGES_REQUESTED** [review]

**Comment by Khareayushh** [comment]
@thorwebdev Resolved prettier in package.json and applied it in quickstart.mdx. Can you please check?

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks 💚

**Comment by thorwebdev** [comment]
Awesome, thanks for sticking with it and getting this across the line @Khareayushh, much appreciated 🙏

**Comment by Khareayushh** [comment]
Big thanks to all @thorwebdev, @laktek, @kevcodez, @saltcod Sirs!

**Pull request merged**